runtime/support/src/lib.rs

```diff
@@ -17,70 +17,53 @@ pub fn panic_fmt() -> ! {
 }
 
 extern "C" {
-	fn ext_print(utf8_data: *const u8, utf8_len: i32);
+	fn ext_print(utf8_data: *const u8, utf8_len: u32);
 	fn ext_print_num(value: u64);
-	fn ext_set_storage(key_data: *const u8, key_len: i32, value_data: *const u8, value_len: i32);
-	fn ext_get_allocated_storage(key_data: *const u8, key_len: i32, written_out: *mut i32) -> *mut u8;
+	fn ext_set_storage(key_data: *const u8, key_len: u32, value_data: *const u8, value_len: u32);
+	fn ext_get_allocated_storage(key_data: *const u8, key_len: u32, written_out: *mut u32) -> *mut u8;
+	fn ext_get_storage_into(key_data: *const u8, key_len: u32, value_data: *mut u8, value_len: u32) -> u32;
+	fn ext_deposit_log(log_data: *const u8, log_len: u32);
 }
 
 pub fn storage(key: &[u8]) -> Vec<u8> {
-	let mut length: i32 = 0;
+	let mut length: u32 = 0;
 	unsafe {
-		let ptr = ext_get_allocated_storage(&key[0], key.len() as i32, &mut length);
+		let ptr = ext_get_allocated_storage(&key[0], key.len() as u32, &mut length);
 		Vec::from_raw_parts(ptr, length as usize, length as usize)
 	}
 }
 
+pub fn storage_into<T: Sized>(key: &[u8]) -> Option<T> {
+	let mut result: T;
+	let size = mem::size_of::<T>();
+	let mut written;
+	unsafe {
```

Comment:
@rphmeier this is pretty nasty (though i'm pretty sure it's safe). maybe you know of a better way of accomplishing the same thing?